backend-lib/src/main/rust/lib.rs

```diff
@@ -374,19 +368,14 @@ pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustDerivation
     let res = panic::catch_unwind(|| {
         let network = parse_network(network_id as u32)?;
         let seed = env.convert_byte_array(seed).unwrap();
-        let account_index = if account_index >= 0 {
-            account_index as u32
-        } else {
-            return Err(format_err!("accountIndex argument must be nonnegative"));
-        };
+        let account_id = account_id_from_jint(account_index)?;
 
-        let account_id = AccountId::from(account_index);
         let ufvk = UnifiedSpendingKey::from_seed(&network, &seed, account_id)
             .map_err(|e| format_err!("error generating unified spending key from seed: {:?}", e))
             .map(|usk| usk.to_unified_full_viewing_key())?;
 
         let (ua, _) = ufvk
-            .find_address(DiversifierIndex::new())
+            .find_address(DiversifierIndex::new(), UnifiedAddressRequest::DEFAULT)
```

Comment:
This is fine for now; I realized while adding the `sapling` feature flag to `zcash_keys` that this default doesn't make sense, because while it's now defined in `zcash_keys`, it's the "default" only in that it is the correct default for the capabilities of `zcash_client_backend` (or really `zcash_client_sqlite`). So I think that this default value will have to move into `zcash_client_sqlite` eventually.